src/bit-systems/waypoint.ts

```diff
@@ -112,6 +116,7 @@ function* moveToSpawnPointJob(world: HubsWorld, characterController: CharacterCo
       snapToNavMesh: true,
       willMaintainInitialOrientation: false
     });
+    initialSpawnHappened = true;
```

Comment:
Can't `initialSpawnHappened` be after the `if` block?

```
  if (spawnPoints.length) {
    const waypoint = spawnPoints[Math.floor(Math.random() * spawnPoints.length)];
    moveToWaypoint(world, waypoint, characterController, true);
  } else {
    console.warn("Could not find any available spawn points, spawning at the origin.");
    characterController.enqueueWaypointTravelTo(new Matrix4().identity(), true, {
      willDisableMotion: false,
      willDisableTeleporting: false,
      snapToNavMesh: true,
      willMaintainInitialOrientation: false
    });
  }
  initialSpawnHappened = true;
```

Reply:
Good call, moved.